### DISCUSSION TITLE
Proposal: Clean up our e2e testing suite.

### DISCUSSION THREAD

**Discussion by kannon92** [armadaproject/armada#1839] [Ideas]
In general, we have a few testing suites.  

1) Unit Testing/ Integration Tests
2) E2E Tests (using kind and docker-run)
3) TestSuite

Currently, TestSuite is only used internally to run CronJob tests in the production environment.  We should have this in our open-source repo running also.  

Some improvements have been thrown around:

1) Use localdev (docker-compose setup) for running our e2e tests.
   - Consolidate our dev environment with what developers are running locally.  
2) Use testsuite for our e2e tests.  
3) Add testsuite as a testing job in our repo.

What do we want in our final testing environment?

**Comment by Sharpz7** [comment]
I think we should also consider writing golang plugins / helpers to do this, like in hashicorps "vault" project does. That way similar code can be used between this, a new localdev setup and the armada setup operator that we may develop.

**Reply by Sharpz7** [reply]
https://github.com/hashicorp/vault/blob/main/plugins/database/postgresql/postgresql_test.go

**Reply by suprjinx** [reply]
we use golang integration tests for e2e already, right? I'm not sure what change is suggested here.
